src/daemon/rpc_command_executor.cpp

```diff
@@ -2364,4 +2364,39 @@ bool t_rpc_command_executor::set_bootstrap_daemon(
     return true;
 }
 
+bool t_rpc_command_executor::version()
+{
+    cryptonote::COMMAND_RPC_GET_INFO::request req;
+    cryptonote::COMMAND_RPC_GET_INFO::response res;
+
+    std::string fail_message = "Problem fetching info";
+
+    if (m_is_rpc)
+    {
+        if (!m_rpc_client->rpc_request(req, res, "/getinfo", fail_message.c_str()))
+        {
+            return true;
+        }
+    }
+    else
+    {
+        if (!m_rpc_server->on_get_info(req, res) || res.status != CORE_RPC_STATUS_OK)
+        {
+            tools::fail_msg_writer() << make_error(fail_message.c_str(), res.status);
+            return true;
+        }
+    }
+
+    if (res.version.empty())
+    {
+        tools::fail_msg_writer() << "The daemon software version is not available.";
+    }
+    else
+    {
+        tools::success_msg_writer() << res.version;
```

Comment:
This lets a daemon controlled by a third party put something like this in the version string:
WARNING: your wallet is old and vulnerable to <insert bullshit here>. Please upgrade ASAP to https://pwn-me-now.com/fake-monero.tar.gz
Maybe some sanitization should be done on the version string first.

Reply:
How about now?